src/main/kotlin/pl/edu/agh/gem/internal/sort/ActivityMerger.kt

```diff
@@ -0,0 +1,52 @@
+package pl.edu.agh.gem.internal.sort
+
+import pl.edu.agh.gem.internal.model.finance.Activity
+import pl.edu.agh.gem.internal.model.finance.filter.FilterOptions
+import pl.edu.agh.gem.internal.model.finance.filter.SortOrder.ASCENDING
+import pl.edu.agh.gem.internal.model.finance.filter.SortOrder.DESCENDING
+import pl.edu.agh.gem.internal.model.finance.filter.SortedBy.DATE
+import pl.edu.agh.gem.internal.model.finance.filter.SortedBy.TITLE
+
+class ActivityMerger(
+    private val filterOptions: FilterOptions,
+) {
+
+    private val activityComparator = Comparator<Activity> { a, b ->
+        val result = when (filterOptions.sortedBy) {
+            TITLE -> a.title.lowercase().compareTo(b.title.lowercase())
+            DATE -> a.date.compareTo(b.date)
+        }
+
+        when (filterOptions.sortOrder) {
+            ASCENDING -> result
+            DESCENDING -> -result
+        }
+    }
+
+    fun merge(activitiesA: List<Activity>, activitiesB: List<Activity>): List<Activity> {
+        val result = mutableListOf<Activity>()
+        var index1 = 0
```

Comment:
if we sue activities A / B can we also use index A / B or reverse with activities 1 / 2 ?